moto/s3/models.py

```diff
@@ -1646,10 +1648,14 @@ def list_buckets(self) -> List[FakeBucket]:
         return list(self.buckets.values())
 
     def get_bucket(self, bucket_name: str) -> FakeBucket:
-        try:
+        if bucket_name in self.buckets:
             return self.buckets[bucket_name]
-        except KeyError:
-            raise MissingBucket(bucket=bucket_name)
+
+        if bucket_name in s3_backends.bucket_accounts:
```

Comment:
Hi @viren-nadkarni! I realize it's been a while.. but what whas the reasoning behind this part?

I guess it made sense to me at the time, but looking at it again, I wonder why we don't just throw an AccessDenied error.

Reply:
Hi @bblommers, it's there to access a bucket that belongs to a different account

Reply:
Right. When I looked at it earlier, an AccessDenied error felt more intuitive.

 But thinking about it a little more, all our cross-account is allowed by default, so this implementation makes more sense. Thanks @viren-nadkarni